gammapy/datasets/map.py

```diff
@@ -1473,6 +1473,41 @@ def stat_sum(self):
         else:
             return cash_sum_cython(counts.ravel(), npred.ravel()) + prior_stat_sum
 
+    def _to_asimov_dataset(self, name=None, models=None):
+        """Create Asimov dataset"""
+        npred = self.npred()
+        data = np.nan_to_num(npred.data, copy=True, nan=0.0, posinf=0.0, neginf=0.0)
```

Comment:
I wonder if `MapDataset.npred()` should directly take some kwargs and perform a `nan_to_num` (defaults with Nones).

Reply:
here I just copied what is done for .fake